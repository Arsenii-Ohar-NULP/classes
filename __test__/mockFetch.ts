export const mockFetch = () => {
    global.fetch = jest.fn();
}