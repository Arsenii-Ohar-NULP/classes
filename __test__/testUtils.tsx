import React, { PropsWithChildren } from 'react';
import { render, renderHook } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import type { RootState, AppStore } from 'pages/redux/store';
// As a basic setup, import your same slice reducers
import { authInitialState, authSlice } from 'pages/redux/auth';
import { classSlice, classesInitialState } from 'pages/redux/classes';

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = { auth: authInitialState, classes: classesInitialState },
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { auth: authSlice.reducer, classes: classSlice.reducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export function renderHookWithProviders(
  ui,
  {
    preloadedState = { auth: authInitialState, classes: classesInitialState },
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { auth: authSlice.reducer, classes: classSlice.reducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...renderHook(() => ui(), { wrapper: Wrapper, ...renderOptions }) };
}
