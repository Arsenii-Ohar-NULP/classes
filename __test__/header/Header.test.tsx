import React from 'react';
import { renderWithProviders } from '__test__/testUtils';
import Header from 'pages/header/header';
import ClassPage from 'pages/class/[id]';

jest.mock('next/router', () => ({
    useRouter: () => ({
      pathname: '/',
      push: jest.fn((url) => console.log(url)),
    }),
  }));

describe('header test', () => {
    it('when no component, should match a snapshot', () => {
        const header = renderWithProviders(<Header currentComponent={null}/>);
        expect(header).toMatchSnapshot();
    })

    it('when ClassPage component, should match a snapshot', () => {
        const header = renderWithProviders(<Header currentComponent={ClassPage}/>);
        expect(header).toMatchSnapshot();
    })
})