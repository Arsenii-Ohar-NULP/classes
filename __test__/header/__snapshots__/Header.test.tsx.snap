// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header test when ClassPage component, should match a snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="p-3 text-white position-relative"
      >
        <div
          class="blur-100"
        >
          <div
            class="header-background"
          />
        </div>
        <div
          class="navbar navbar-expand-lg"
        >
          <div
            class="navbar-brand rounded me-2"
          >
            <div
              class="d-flex justify-content-center align-items-center gap-1 bg-dark p-2 rounded"
            >
              <img
                alt="Classes Logo"
                class="border rounded-3 p-1 m-0 fill-light classes-pic"
                data-nimg="1"
                decoding="async"
                height="32"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
                style="color: transparent;"
                width="32"
              />
              <h3
                class="m-0 align-middle text-primary"
              >
                Classes
              </h3>
            </div>
          </div>
          <button
            aria-controls="navbarNav"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="navbar-toggler"
            data-bs-target="#navbarNav"
            data-bs-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse col"
            id="navbarNav"
          >
            <ul
              class="nav col-12 col-lg-auto me-lg-auto mb-2 justify-content-center mb-md-0"
            >
              <li>
                <a
                  class="nav-link px-2 text-dark"
                  href="/classes"
                >
                  <b>
                    Home
                  </b>
                </a>
              </li>
            </ul>
            <div
              class="text-center text-lg-end"
            >
              <div
                class="profile-pic-size d-inline"
              >
                <div
                  class="spinner-grow text-primary align-middle mx-2 p-3 spinner-size"
                  role="status"
                >
                  <span
                    class="visually-hidden"
                  >
                    Loading...
                  </span>
                </div>
              </div>
              <button
                class="btn btn-primary"
                type="button"
              >
                Logout
              </button>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="p-3 text-white position-relative"
    >
      <div
        class="blur-100"
      >
        <div
          class="header-background"
        />
      </div>
      <div
        class="navbar navbar-expand-lg"
      >
        <div
          class="navbar-brand rounded me-2"
        >
          <div
            class="d-flex justify-content-center align-items-center gap-1 bg-dark p-2 rounded"
          >
            <img
              alt="Classes Logo"
              class="border rounded-3 p-1 m-0 fill-light classes-pic"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <h3
              class="m-0 align-middle text-primary"
            >
              Classes
            </h3>
          </div>
        </div>
        <button
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarNav"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse col"
          id="navbarNav"
        >
          <ul
            class="nav col-12 col-lg-auto me-lg-auto mb-2 justify-content-center mb-md-0"
          >
            <li>
              <a
                class="nav-link px-2 text-dark"
                href="/classes"
              >
                <b>
                  Home
                </b>
              </a>
            </li>
          </ul>
          <div
            class="text-center text-lg-end"
          >
            <div
              class="profile-pic-size d-inline"
            >
              <div
                class="spinner-grow text-primary align-middle mx-2 p-3 spinner-size"
                role="status"
              >
                <span
                  class="visually-hidden"
                >
                  Loading...
                </span>
              </div>
            </div>
            <button
              class="btn btn-primary"
              type="button"
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`header test when no component, should match a snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
